src/lib/basic_types/base_table.rs

```diff
@@ -0,0 +1,105 @@
+/// When translator meets a base at a given
+/// location couter (xtr) it'll ask the state
+/// if a base is available for xtr
+
+use std::u32;
+use parking_lot::RwLock;
```

Comment:
we dont have multi threading here on any level so explain, mutexes or what ever you used here introduce new level of bugs and there should always be better ways to solve it